recipes-extended/xen/files/libxl-openxt-qemu-args.patch

```diff
@@ -0,0 +1,188 @@
+     return libxl__strdup(gc, "no");
+ #else
+-    return libxl__sprintf(gc, "%s/qemu-ifup", libxl__xen_script_dir_path());
++    return libxl__sprintf(gc, "%s/qemu/qemu-ifup", libxl__xen_script_dir_path());
+ #endif
+ }
+ 
+@@ -458,14 +458,7 @@ static int libxl__build_device_model_arg
+         if (libxl_defbool_val(vnc->findunused)) {
+             flexarray_append(dm_args, "-vncunused");
+         }
+-    } else
+-        /*
+-         * VNC is not enabled by default by qemu-xen-traditional,
+-         * however passing -vnc none causes SDL to not be
+-         * (unexpectedly) enabled by default. This is overridden by
+-         * explicitly passing -sdl below as required.
+-         */
+-        flexarray_append_pair(dm_args, "-vnc", "none");
++    } /* OpenXT: no else here, we don't support "-vnc none" */
+ 
+     if (sdl) {
+         flexarray_append(dm_args, "-sdl");
+@@ -525,7 +518,7 @@ static int libxl__build_device_model_arg
+ 
+         switch (b_info->u.hvm.vga.kind) {
+         case LIBXL_VGA_INTERFACE_TYPE_STD:
+-            flexarray_append(dm_args, "-std-vga");
++            flexarray_append_pair(dm_args, "-vga", "std");
+             break;
+         case LIBXL_VGA_INTERFACE_TYPE_CIRRUS:
+             break;
+@@ -606,10 +599,9 @@ static int libxl__build_device_model_arg
+                 ioemu_nics++;
+             }
+         }
+-        /* If we have no emulated nics, tell qemu not to create any */
+-        if ( ioemu_nics == 0 ) {
+-            flexarray_vappend(dm_args, "-net", "none", NULL);
+-        }
++
++        /* OpenXT: We don't support -net none, adding nothing if there's 0 nic */
++
+         if (libxl_defbool_val(b_info->u.hvm.gfx_passthru)) {
+             flexarray_append(dm_args, "-gfx_passthru");
+         }
+@@ -737,19 +729,22 @@ static int libxl__build_device_model_arg
+                       "-xen-domid",
+                       libxl__sprintf(gc, "%d", guest_domid), NULL);
+ 
+-    /* There is currently no way to access the QMP socket in the stubdom */
+     if (!is_stubdom) {
+-        flexarray_append(dm_args, "-chardev");
++        flexarray_append(dm_args, "-qmp");
+         flexarray_append(dm_args,
+-                         libxl__sprintf(gc, "socket,id=libxl-cmd,"
+-                                        "path=%s/qmp-libxl-%d,server,nowait",
++                         libxl__sprintf(gc, "unix:%s/qmp-libxl-%d,server,nowait",
+                                         libxl__run_dir_path(), guest_domid));
+-
+-        flexarray_append(dm_args, "-no-shutdown");
+-        flexarray_append(dm_args, "-mon");
+-        flexarray_append(dm_args, "chardev=libxl-cmd,mode=control");
++    } else {
++        /* OpenXT: We have V4V qmp, proxied by the qmp_helper */
++        flexarray_append_pair(dm_args, "-qmp", "v4v");
+     }
+ 
++    /* OpenXT: more options specific to us */
++    flexarray_append_pair(dm_args, "-device", "xenmou");
++    /* FIXME: -xen-acpi-pm seems to upset non-stubdom guests */
```

Comment:
What is the problem here? APCI PM feature should (and used to) work w/ or w/o stubdoms. If something is broken we need a ticket for it.